vsg/cmd_line_args.py

```diff
@@ -14,44 +12,65 @@
 from . import severity
 from . import version
 from . import vhdlFile
+import argparse
+from .CustomArgumentParser import CustomArgumentParser
 
 
-def parse_command_line_arguments():
-    '''Parses the command line arguments and returns them.'''
+def __check_strict_positive(value):
+    iValue: int = int(value)
+    if iValue <= 0:
+        raise argparse.ArgumentTypeError(f"{value} is an invalid value for number of jobs")
+    return iValue
 
-    def check_strict_positive(value):
-        iValue = int(value)
-        if iValue <= 0:
-            raise argparse.ArgumentTypeError(f"{value} is an invalid value for number of jobs")
-        return iValue
 
-    parser = argparse.ArgumentParser(
-      prog='VHDL Style Guide (VSG)',
-      description='''Analyzes VHDL files for style guide violations.
+def __is_valid_file(value: str) -> str:
+    """
+    Check an argument in argparse to be a path to an existing file
+    :param value: String path to analyze.
+    :return:
+    """
+    if not os.path.isfile(value):
+        raise argparse.ArgumentTypeError(f"The file {value} does not exist.")
+    return value
+
+
+def parse_command_line_arguments():
+    """Parses the command line arguments and returns them."""
+
+    parser = CustomArgumentParser(
+        prog='VHDL Style Guide (VSG)',
+        description='''Analyzes VHDL files for style guide violations.
                    Reference documentation is located at:
                    http://vhdl-style-guide.readthedocs.io/en/latest/index.html''')
 
-    parser.add_argument('-f', '--filename', nargs='+', help='File to analyze')
+    parser.add_argument('-f', '--filename', type=__is_valid_file, nargs='+', help='File to analyze')
     parser.add_argument('-lr', '--local_rules', help='Path to local rules')
     parser.add_argument('-c', '--configuration', nargs='+', help='JSON or YAML configuration file(s)')
```

Comment:
Should we add the `type=__is_valid_file` to this argument also?

Reply:
Yes, we should.